services/NotificationAgent/src/notification_agent/messages.clj

```diff
@@ -54,35 +55,16 @@
   (db/insert-notification
    (or type "analysis") username subject created-date (cheshire/encode msg)))
 
-(defn- send-msg-to-recipient
-  "Forawards a message to a single recipient."
-  [url msg]
-  (log/debug "sending message to" url)
-  (try
-    (client/post url {:body msg})
-    (catch IOException e
-      (log/error e "unable to send message to" url))))
-
-(defn send-msg
-  "Forwards a message to zero or more recipients."
-  ([uuid msg]
-     (let [recipients (notification-recipients)
-           msg        (cheshire/encode (reformat-message uuid msg))]
-       (log/debug "forwarding message to" (count recipients) "recipients")
-       (dorun (map #(send-msg-to-recipient % msg) recipients))))
-  ([msg]
-     (send-msg nil msg)))
-
 (defn persist-and-send-msg
   "Persists a message in the notification database and sends it to any receivers
    and returns the state object."
-  [msg]
+  [{:keys [user] :as msg}]
   (let [uuid          (persist-msg msg)
         email-request (:email_request msg)]
     (log/debug "UUID of persisted message:" uuid)
     (when-not (nil? email-request)
       (.start (Thread. #(send-email-request uuid email-request))))
-    (send-msg uuid msg)))
+    (amqp/publish-msg user msg)))
```

Comment:
(unrelated)

it'll be really nice when we can also send the email request as an AMQP message here instead of needing to start extra threads!